gems/sorbet-runtime/lib/types/props/decorator.rb

```diff
@@ -174,7 +174,9 @@ def prop_get_logic(instance, prop, value)
     .checked(:never)
   end
   def prop_get(instance, prop, rules=prop_rules(prop))
-    val = instance.instance_variable_get(rules[:accessor_key]) if instance.instance_variable_defined?(rules[:accessor_key])
+    # `instance_variable_get` will return nil if the variable doesn't exist
+    # which is what we want to have happen for the logic below.
+    val = instance.instance_variable_get(rules[:accessor_key])
```

Comment:
Unfortunately, this causes warnings if warnings are enabled.

Reply:
It causes warnings if they're enabled, but only if you're directly calling `prop_get`. Is that a supported API? At that point I'd wonder if it's worth just having an internal one that won't deal with calling that to avoid the extra calls.